packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -242,34 +338,47 @@ exports.reconciliationReportForGranules = reconciliationReportForGranules;
 /**
  * Compare the holdings in CMR with Cumulus' internal data store, report any discrepancies
  *
+ * @param {Object} bucketsConfig - bucket configuration object
  * @returns {Promise<Object>} a reconciliation report
  */
-async function reconciliationReportForCumulusCMR() {
+async function reconciliationReportForCumulusCMR(bucketsConfig) {
   const collectionReport = await reconciliationReportForCollections();
   const collectionsInCumulusCmr = {
-    okCollectionCount: collectionReport.okCollections.length,
+    okCount: collectionReport.okCollections.length,
     onlyInCumulus: collectionReport.onlyInCumulus,
     onlyInCmr: collectionReport.onlyInCmr
   };
 
   // create granule report for collections in both Cumulus and CMR
   const promisedGranuleReports = collectionReport.okCollections.map((collectionId) =>
-    reconciliationReportForGranules(collectionId));
-  const granuleReports = await Promise.all(promisedGranuleReports);
+    reconciliationReportForGranules(collectionId, bucketsConfig));
+  const granuleAndFilesReports = await Promise.all(promisedGranuleReports);
 
   const granulesInCumulusCmr = {
-    okGranuleCount: 0,
+    okCount: 0,
+    onlyInCumulus: [],
+    onlyInCmr: []
+  };
+  const filesInCumulusCmr = {
+    okCount: 0,
     onlyInCumulus: [],
     onlyInCmr: []
   };
-  granuleReports.forEach((granuleReport) => {
-    granulesInCumulusCmr.okGranuleCount += granuleReport.okGranulesInDb.length;
-    granulesInCumulusCmr.onlyInCumulus = granulesInCumulusCmr
-      .onlyInCumulus.concat(granuleReport.onlyInCumulus);
-    granulesInCumulusCmr.onlyInCmr = granulesInCumulusCmr.onlyInCmr.concat(granuleReport.onlyInCmr);
+  granuleAndFilesReports.forEach((granuleFileReport) => {
```

Comment:
Seems like a `.reduce()` operation might be appropriate here and more functional, but it's not strictly necessary

Reply:
Good suggestion.  I updated the code to use `reduce()`